dds/DCPS/XTypes/TypeObject.cpp

```diff
@@ -322,7 +322,287 @@ bool has_type_object(const TypeIdentifier& ti)
 }
 
 namespace {
+void compute_dependencies(const TypeMap& type_map,
+                          const MinimalAnnotationParameter& type,
+                          OPENDDS_SET(TypeIdentifier)& dependencies);
+void compute_dependencies(const TypeMap& type_map,
+                          const CompleteAnnotationParameter& type,
+                          OPENDDS_SET(TypeIdentifier)& dependencies);
+
+void compute_dependencies(const TypeMap& type_map,
+                          const CommonAliasBody& type,
+                          OPENDDS_SET(TypeIdentifier)& dependencies);
+
```

Comment:
I think we just need to rearrange these `compute_dependencies` methods' definitions to resolve any compile errors instead of adding all their declarations in front.

Reply:
This turned out to be a lot more complicated than expected.

Essentially:
Because we are in an anonymous namespace with no .h or separate declaration, every type in a compute dependency that could be a sequence would need to know of every type that could be in that sequence.   A quick example is having 2 types: Type A and Type B.

Type A can be a sequence and can contain both type A and type B in that sequence.
Type B can be a sequence and can contain both type A and type B in that sequence.

The template tries to resolve this but what happens is that because it matters where the template is defined, the template needs to be above anything that can be a sequence but also below anything that can be in a sequence.  This means it would both need to be above A and B, and below A and B in order to compile.

The solution to this is to declare only the template, then implement the template at the end of the anonymous namespace.

Reply:
Seems like you mean there are some types that mutually depend on each other. Can you give a specific example for such types A and B?